src/cairo/bootloader.md

```diff
@@ -1 +1,199 @@
+
+### Step 1: A Cairo verifier program
+
+First, we need to understand what a Cairo verifier program is.
+
+A Cairo verifier is a Cairo program that verifies a Cairo proof, and in this case, it will only verify a proof generated using the "simple" bootloader. Below is an example of what running a Cairo verifier program looks like.
+
+<div style="text-align: center;">
+    <img src="cairo_verifier_example.png" alt="Cairo verifier example" width="70%">
+</div>
+
+Note that when the Cairo verifier verifies the simple bootloader proof, it includes in the output the hash of the simple bootloader program and the hash of the outputs of the simple bootloader. This means that it is possible to commit to the output of the simple bootloader using the Cairo verifier and open it later.
+
+### Step 2: Iteratively creating proofs
+
+Looking closely at diagram above, you can see that we can tweak it a little bit to create a perfect loop. Since the simple bootloader can run any arbitrary program and the Cairo verifier requires a proof of the simple bootloader, we can do something like the following:
+
+<div style="text-align: center;">
+    <img src="perfect_loop.png" alt="Perfect loop" width="40%">
+</div>
+
+Of course, there is no point in just repeating this loop, but if we tweak this just a little bit again, we can do something more interesting: creating a proof that both verifies an existing proof and proves a new program. We'll call this "iteratively creating proofs", which is the core of what the "advanced" bootloader does.
+
+<div style="text-align: center;">
+    <img src="iteratively_creating_proofs.png" alt="Iteratively creating proofs" width="60%">
+</div>
+
+This might still not seem too useful, but it unlocks a new use-case for users who want to create proofs of programs as they come, not just when they have a fixed set. For example, L2 nodes that need to create a single proof of programs that are transmitted over a 15-minute period do not need to wait for the whole 15 minutes and create a proof at the last minute, which would take a long time if the number of programs to be proven are large. Instead, they can create a proof after the first 1 minute, and for the next 14 times every minute they can create a new proof that recursively proves the previous proof and proves the new programs.
+
+This might still not seem too useful, but it unlocks a new use-case for users who want to create proofs of programs as they come, not just when they have a fixed set. For example, L2 nodes that need to create a single proof of programs that are transmitted over a 15-minute period do not need to wait for the whole 15 minutes and create a proof at the last minute, which would take a long time if the number of programs to be proven are large. Instead, they can create a proof after the first 1 minute, and for the next 14 times every minute they can create a new proof that recursively proves the previous proof and proves the new programs.
```

Comment:
It also overcomes memory limits of proving: if you simply bundle up everything (one giant bootloader proof), then you will run out of resources sooner rather than later :)